tests/unit/externalRequests.js

```diff
@@ -17,8 +17,249 @@ const handleRequestError = externalRequests.__get__('handleRequestError')
 const addRequestQueryParameters = externalRequests.__get__(
   'addRequestQueryParameters'
 )
+const validateRequestStatusCode = externalRequests.__get__(
+  'validateRequestStatusCode'
+)
+const performRequests = externalRequests.__get__('performRequests')
+const prepareLookupRequests = externalRequests.__get__('prepareLookupRequests')
+const prepareRequestConfig = externalRequests.__get__('prepareRequestConfig')
 
 tap.test('External Requests', {autoend: true}, t => {
+  t.test('validateRequestStatusCode', {autoend: true}, t => {
+    t.test('should return false if the status is not allowed', t => {
+      const allowedStatuses = [200, 202]
+      const status = 500
+
+      const valid = validateRequestStatusCode(allowedStatuses)(status)
+      t.equals(valid, false)
+      t.end()
+    })
+
+    t.test('should return true if status is allowed', t => {
+      const allowedStatuses = [200, 202]
+      const status = 200
+
+      const valid = validateRequestStatusCode(allowedStatuses)(status)
+      t.equals(valid, true)
+      t.end()
+    })
+
+    t.test('should return true if status is allowed (wildcard status)', t => {
+      const allowedStatuses = ['2xx', 400]
+      const status = 202
+
+      const valid = validateRequestStatusCode(allowedStatuses)(status)
+      t.equals(valid, true)
+      t.end()
+    })
+  })
+
+  t.test('performRequests', {autoend: true}, t => {
+    t.test('should create orchestration object', t => {
+      const ctx = {}
+      const requests = []
+
+      performRequests(requests, ctx)
+      t.equals(!ctx.orchestrations, false)
+      t.end()
+    })
+
+    t.test('should create orchestrations array', t => {
+      const ctx = {}
+      const requests = []
+
+      performRequests(requests, ctx)
+      t.equals(!ctx.orchestrations, false)
+      t.end()
+    })
+
+    t.test(
+      'should throw error and create orchestration when one lookup fails( request to server that is down)',
+      async t => {
+        const url = 'http://localhost:8000/'
+        const url2 = 'http://localhost:9000/'
+        const method = 'PUT'
+        const id = '1233243'
+
+        const requests = [
+          {
+            config: {
+              url: url,
+              method: method
+            },
+            id: id
+          },
+          {
+            config: {
+              url: url2,
+              method: method
+            },
+            id: id
+          }
+        ]
+
+        const ctx = {
+          state: {
+            metaData: {
+              requests: {}
+            }
+          },
+          request: {
+            header: {
+              'x-openhim-transactionid': '1232244'
+            }
+          }
+        }
+
+        try {
+          await Promise.all(performRequests(requests, ctx))
+        } catch (error) {
+          t.equals(ctx.orchestrations.length, 1)
+          t.match(error.message, /ECONNREFUSED/)
```

Comment:
@bradsawadye 

The message that you're looking for could include a bit more detail - helps to understand where it's testing

```suggestion
          t.match(
            error.message,
            /No response from lookup '1233243'. connect ECONNREFUSED/
          )
```